web/source/kmwembedded.ts

```diff
@@ -326,6 +332,20 @@ namespace com.keyman.text {
     return x+','+y+','+w+','+h;
   };
 
+  keymanweb['showGlobeHint'] = function(text: string, onAutodismissal?: () => void) {
+    // Ensure localized text is properly in-place.
+    if((keymanweb as KeymanBase).osk?.vkbd) {
+      let hint = (keymanweb as KeymanBase).osk.vkbd.globeHint as com.keyman.osk.embedded.GlobeHint;
+      hint.text = text;
+    }
+
+    keymanweb.osk?.vkbd?.globeHint?.show(keymanweb.osk.vkbd.currentLayer.globeKey.btn, onAutodismissal);
```

Comment:
A few niggles here:
* I don't know why you are casting `keymanweb as KeymanBase` in some places but not others?
* You are casting `globeHint` as `GlobeHint` on line 338 but not on line 342.
* You are using `globeHint?.` on line 342 but on 339 you just use `hint.`.

```suggestion
    const vkbd = keymanweb?.osk?.vkbd;
    if(!vkbd?.globeHint) {
      return;
    }
    vkbd.hint.text = text;
    vkbd.hint.show(vkbd.currentLayer.globeKey.btn, onAutodismissal);
```

* And finally, is `keymanweb.osk.vkbd.currentLayer.globeKey.btn` guaranteed to always exist given all the other nullish tests you have above?